File: scratch_update.py
import re

with open(r'C:\Users\hajal\.gemini\antigravity-ide\brain\a5be9aa3-700a-47f1-8aa5-74bd8a95ab9e\scratch\extracted.html', 'r', encoding='utf-8') as f:
    new_html = f.read()

with open(r'd:\HAJA LUZ\Antigravity\Dados_IA\scratch\oraculum-saas\public\index.html', 'r', encoding='utf-8') as f:
    index_html = f.read()

body_match = re.search(r'<body>(.*?)</body>', new_html, re.DOTALL)
new_body = body_match.group(1) if body_match else ""

head_match = re.search(r'<head>(.*?)</head>', new_html, re.DOTALL)
new_head = head_match.group(1) if head_match else ""

style_match = re.search(r'<style>.*?</style>', new_head, re.DOTALL)
style_content = style_match.group(0) if style_match else ""
script_tags = '\n<script src="https://cdnjs.cloudflare.com/ajax/libs/three.js/r128/three.min.js"></script>\n'

# Scope body styles to #auth-gate-container so dashboard is not affected
style_content = style_content.replace('html, body {', '#auth-gate-container {')

# Hook up form to existing app login
new_body = new_body.replace('onsubmit="handleLogin(event)"', 'onsubmit="window.executarLogin(event)" id="form-login"')
new_body = new_body.replace('id="auth-email"', 'id="login-email"')
new_body = new_body.replace('id="auth-password"', 'id="login-password"')
new_body = new_body.replace('class="btn-login-submit"', 'class="btn-login-submit" id="btn-submit-login"')
new_body = new_body.replace('</form>', '  <input type="checkbox" id="remember-me" checked style="display: none;">\n    </form>')

# Remove dummy handleLogin function
new_body = re.sub(r'function handleLogin\(e\)\s*\{.*?\}(?=\s*// =+)', '', new_body, flags=re.DOTALL)

# Wrap new elements
wrapped_new_content = f"""  <div id="auth-gate-container" class="fixed inset-0 z-50 bg-[#010403] relative overflow-hidden" style="display: flex !important;">
{script_tags}
{style_content}
{new_body}
  </div>"""

# Replace existing auth-gate-container
replace_pattern = r'<div id="auth-gate-container"[\s\S]*?(?=<!-- MAIN DASHBOARD CONTAINER)'
index_html = re.sub(replace_pattern, wrapped_new_content + '\n\n  ', index_html, count=1)

with open(r'd:\HAJA LUZ\Antigravity\Dados_IA\scratch\oraculum-saas\public\index.html', 'w', encoding='utf-8') as f:
    f.write(index_html)

File: scratch_update.js
const fs = require('fs');

const extractedPath = 'C:\\Users\\hajal\\.gemini\\antigravity-ide\\brain\\a5be9aa3-700a-47f1-8aa5-74bd8a95ab9e\\scratch\\extracted.html';
const indexPath = 'd:\\HAJA LUZ\\Antigravity\\Dados_IA\\scratch\\oraculum-saas\\public\\index.html';

const newHtml = fs.readFileSync(extractedPath, 'utf-8');
let indexHtml = fs.readFileSync(indexPath, 'utf-8');

const bodyMatch = newHtml.match(/<body>([\s\S]*?)<\/body>/);
let newBody = bodyMatch ? bodyMatch[1] : '';

const headMatch = newHtml.match(/<head>([\s\S]*?)<\/head>/);
const newHead = headMatch ? headMatch[1] : '';

const styleMatch = newHead.match(/<style>[\s\S]*?<\/style>/);
let styleContent = styleMatch ? styleMatch[0] : '';
const scriptTags = '\n<script src="https://cdnjs.cloudflare.com/ajax/libs/three.js/r128/three.min.js"></script>\n';

styleContent = styleContent.replace('html, body {', '#auth-gate-container {');

newBody = newBody.replace('onsubmit="handleLogin(event)"', 'onsubmit="window.executarLogin(event)" id="form-login"');
newBody = newBody.replace('id="auth-email"', 'id="login-email"');
newBody = newBody.replace('id="auth-password"', 'id="login-password"');
newBody = newBody.replace('class="btn-login-submit"', 'class="btn-login-submit" id="btn-submit-login"');
newBody = newBody.replace('</form>', '  <input type="checkbox" id="remember-me" checked style="display: none;">\n    </form>');

newBody = newBody.replace(/function handleLogin\(e\)[\s\S]*?(?=\s*\/\/ =+)/, '');

const wrappedNewContent = `  <div id="auth-gate-container" class="fixed inset-0 z-50 bg-[#010403] relative overflow-hidden" style="display: flex !important;">
${scriptTags}
${styleContent}
${newBody}
  </div>`;

const replacePattern = /<div id="auth-gate-container"[\s\S]*?(?=<!-- MAIN DASHBOARD CONTAINER)/;
indexHtml = indexHtml.replace(replacePattern, wrappedNewContent + '\n\n  ');

fs.writeFileSync(indexPath, indexHtml, 'utf-8');
console.log('Update successful');
